packages/kbn-eslint-plugin-eslint/rules/no_restricted_paths.js

```diff
@@ -97,11 +97,20 @@ module.exports = {
     }
 
     function checkForRestrictedImportPath(importPath, node) {
-      const absoluteImportPath = resolve(importPath, context);
-      if (!absoluteImportPath) return;
+      let absoluteImportPath;
+      const [one, two, ...rest] = importPath.split('/');
+      if (one === 'ui') {
+        absoluteImportPath = path.join(
+          basePath,
+          `src/legacy/${one}/public/${two}${rest.length ? `/${rest.join('/')}` : ''}`
```

Comment:
could you add a couple of tests in https://github.com/restrry/kibana/blob/ede7e6da750d4ae471ede364018d336a3d9192c8/packages/kbn-eslint-plugin-eslint/rules/__tests__/no_restricted_paths.js

for example, later we can add `webpack alias` parameters to keep all the declarations in .eslintrc
```js
// similar to create_jest_config.js
moduleNameMapper: {
  '^ui/(.*)': 'src/legacy/ui/public/$1',
}
```